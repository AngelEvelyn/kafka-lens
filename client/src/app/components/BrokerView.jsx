import React from 'react';
import PropTypes from 'prop-types';
import BrokerHeader from './BrokerHeader';
import BrokerTopicsView from './BrokerTopicsView';
import '../css/BrokerView.scss';

const brokerColors = {
  Red: '#DC143C',
  Green: '#90EE90',
};

const BrokerView = ({ isAlive, openSideBar, brokerId, brokerURI, topics }) => {
  let backgroundColor;
  if (isAlive === false) {
    backgroundColor = brokerColors.Red;
  } else {
    backgroundColor = brokerColors.Green;
  }

  return (
    <div
      className="brokerView"
      role="button"
      tabIndex="0"
      onKeyPress={() => openSideBar(brokerId, topics)}
      onClick={() => openSideBar(brokerId, topics)}
      key={brokerId}
      style={{ backgroundColor }}
    >
      <BrokerHeader brokerID={brokerId} brokerURI={brokerURI} />
      <BrokerTopicsView topics={topics} />
    </div>
  );
};

export default BrokerView;

BrokerView.propTypes = {
  isAlive: PropTypes.bool.isRequired,
  openSideBar: PropTypes.func.isRequired,
  brokerId: PropTypes.number.isRequired,
  brokerURI: PropTypes.string.isRequired,
  topics: PropTypes.arrayOf(PropTypes.object).isRequired,
};
