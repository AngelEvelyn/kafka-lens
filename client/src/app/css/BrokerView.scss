.brokerView {
  // display: grid;
  // align-items: center;
  // justify-content: space-between;
  padding: 20px;
  opacity: 0.5;
}

.brokerView:hover {
  opacity: 1;
}
