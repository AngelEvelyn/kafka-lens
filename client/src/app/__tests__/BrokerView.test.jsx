import React from 'react';
import { shallow } from 'enzyme';

import BrokerView from '../components/BrokerView';

describe('BrokerView.jsx unit tests', () => {
  let wrapper;

  const props = {
    isAlive: false,
    openSideBar: jest.fn(),
    brokerId: 1,
    brokerURI: 'localhost:9092',
    topics: [],
  };

  const brokerColors = {
    Red: '#DC143C',
    Green: '#90EE90',
  };

  // { isAlive, openSideBar, brokerId, brokerURI, topics }

  beforeAll(() => {
    wrapper = shallow(<BrokerView {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // test if broker header renders
  it('Should render <BrokerHeader/>', () => {
    expect(wrapper.exists('BrokerHeader')).toEqual(true);
  });

  // test if broker topic view renders
  it('Should render <BrokerTopicsView/>', () => {
    expect(wrapper.exists('BrokerTopicsView')).toEqual(true);
  });

  it('If Broker is inactive, then color should be red', () => {
    expect(wrapper.find('.brokerView').get(0).props.style).toHaveProperty(
      'backgroundColor',
      brokerColors.Red,
    );
  });

  it('If Broker is active, then color should be green', () => {
    wrapper.setProps({
      isAlive: true,
    });
    expect(wrapper.find('.brokerView')).toHaveLength(1);
    expect(wrapper.find('.brokerView').get(0).props.style).toHaveProperty(
      'backgroundColor',
      brokerColors.Green,
    );
  });
});
