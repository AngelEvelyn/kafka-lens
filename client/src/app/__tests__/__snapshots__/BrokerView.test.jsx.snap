// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrokerView.jsx unit tests Should render without errors 1`] = `
<div
  className="brokerView"
  key="1"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "backgroundColor": "#DC143C",
    }
  }
  tabIndex="0"
>
  <BrokerHeader
    brokerID={1}
    brokerURI="localhost:9092"
  />
  <BrokerTopicsView
    topics={Array []}
  />
</div>
`;
